lib/sync.js

```diff
@@ -0,0 +1,51 @@
+import VP from "./vp";
+import {
+  getFlowsByStatus,
+  updateParliamentFlowStatus,
+} from "./parliament-flow";
+import { PARLIAMENT_FLOW_STATUSES, VP_PARLIAMENT_FLOW_STATUSES } from "../config";
+
+/**
+ * For all flows in Kaleidos that have a given status, it checks for
+ * status updates on the VP API and updates their status in
+ * Kaleidos accordingly.
+ */
+async function syncFlowsByStatus(statuses) {
+  const pendingFlows = await getFlowsByStatus(statuses);
+  await Promise.all(
+    pendingFlows.map(async (flow) => {
+      let statuses;
+      try {
+        statuses = await VP.getStatusForFlow(flow.parliamentId);
+      } catch (error) {
+        console.error(error);
+      }
+      if (!statuses?.statussen) {
+        await updateParliamentFlowStatus(
+          flow.uri,
+          PARLIAMENT_FLOW_STATUSES.VP_ERROR,
+          true
+        );
+        console.warn(
+          `Invalid response from status API for flow with uri ${flow.uri}`
+        );
+        return;
+      }
+      if (
+        statuses.statussen.find(
+          (statusChange) =>
+            statusChange.status === VP_PARLIAMENT_FLOW_STATUSES.BEING_HANDLED
```

Comment:
**question**: Have you been able to test this? Locally I can't get the API to respond with anything other than `{ statussen: null }`, so all my flows get the error status. Or is this code based on a convention agreed with (but maybe not implemented yet by) the VP?

Reply:
Yes, I tested this with the case with id 1729615, which does give a response.